commons-db/src/main/java/com/palantir/nexus/db/sql/SqlConnectionHelper.java

```diff
@@ -488,7 +489,31 @@ void loadTempIds(Connection c, Iterable<Long> tempIds, String tableName, ClearSt
             }
             Iterable<Object[]> args = idsToArguments(tempIds);
             String sqlInsert = TextUtils.format(SQL_INSERT_INTO_PT_TEMP_IDS, tableName);
-            insertManyUnregisteredQuery(c, sqlInsert, args); // dynamic query
+            try {
+                insertManyUnregisteredQuery(c, sqlInsert, args); // dynamic query
+            } catch (PalantirSqlException e) {
+                assert checkRepeatedInsert(e, c, tableName, tempIds);
+                throw e;
+            }
+        }
+    }
+
+    /**
+     * Check if the temp table has this exact set of tempIds already, for debugging purposes.
+     * This should be behind an assert so it does not run in production.
```

Comment:
this is not a guarantee shared by oss consumers, but I don't care if the workflow where someone was already throwing exceptions is slower than previously